astropy/io/fits/column.py

```diff
@@ -1389,16 +1387,13 @@ def __getattr__(self, name):
 
         Implements for example self.units, self.formats, etc.
         """
-
         cname = name[:-1]
         if cname in KEYWORD_ATTRIBUTES and name[-1] == 's':
-            attr = []
-            for col in self:
-                val = getattr(col, cname)
-                if val is not None:
-                    attr.append(val)
-                else:
-                    attr.append('')
+            attr = [getattr(col, cname) for col in self.columns]
+            if None in attr:
```

Comment:
@pllim I know cases where a `list`-constructor call is faster than a list comprehension but I've never seen an explicit loop be faster than an equivalent comprehension. 
Would you mind giving me an example? I'm always eager to learn more about optimization and I often use list comprehensions.

I'd be interested which approach is faster for @saimn because the performance really depends on if there is a column without name or if there isn't.